fast64_internal/sm64/sm64_objects.py

```diff
@@ -1403,6 +1420,784 @@ def draw_act(self, obj, layout, value):
+            return self.context_obj or bpy.context.active_object
+        else:
+            return self.col_object or self.gfx_object or bpy.context.active_object
+
+    @property
+    def group_num(self):
+        if self.group_name == "common0":
+            return 1
+        else:
+            return int(self.group_name.removeprefix("group")) + 1
+
+    @property
+    def obj_name_col(self):
+        if self.export_all_selected:
+            return self.col_object.name
+        if not self.object_name and not self.col_object:
+            return ""
+        else:
+            return self.object_name or self.col_object.name
+
+    @property
+    def obj_name_gfx(self):
+        if self.export_all_selected:
+            return self.gfx_object.name
+        if not self.object_name and not self.gfx_object:
+            return ""
+        else:
+            return self.object_name or self.gfx_object.name
+
+    @property
+    def obj_name_bhv(self):
+        if self.export_all_selected:
+            return self.bhv_object.name
+        if not self.object_name and not self.bhv_object:
+            return ""
+        else:
+            return self.object_name or self.bhv_object.name
+
+    @property
+    def bhv_name(self):
+        return "bhv" + "".join([word.title() for word in toAlnum(self.obj_name_bhv).split("_")])
+
+    @property
+    def geo_name(self):
+        return f"{toAlnum(self.obj_name_gfx)}_geo"
+
+    @property
+    def collision_name(self):
+        return f"{toAlnum(self.obj_name_col)}_collision"
+
+    @property
+    def model_id_define(self):
+        return f"MODEL_{toAlnum(self.obj_name_gfx)}".upper()
+
+    @property
+    def export_level_name(self):
+        if self.level_name == "custom":
+            return self.custom_export_name
+        return self.level_name
+
+    def draw_generic_options(self, layout):
+        split = layout.split(factor=0.333)
+        col = split.column()
+        if self.export_col:
+            box = col.box()
+            box.prop(self, "export_col")
+            box.prop(self, "include_children")
+            box.prop(self, "export_rooms")
+        else:
+            col.prop(self, "export_col")
+        split.prop(self, "export_gfx")
+        if not self.export_all_selected:
+            split.prop(self, "export_bhv")
+        row = layout.row()
+        row.prop(self, "export_all_selected")
+        row.prop(self, "export_script_loads")
+        self.draw_obj_name(layout)
+
+    def draw_level_path(self, layout):
+        if self.export_header_type == "Custom":
+            export_path = f"{toAlnum(self.custom_export_path)}/"
+        else:
+            export_path = f"/levels/{toAlnum(self.level_name)}/"
+        layout.label(text=f"Level export location is: {export_path}")
+
+    def draw_gfx_names(self, layout):
+        if self.export_header_type == "Actor":
+            layout.label(text=f"Objects will export to /actors/{toAlnum(self.obj_name_gfx)}(.c, .h, _geo.c)")
```

Comment:
No, it exports to the name of the mesh of the armature, the mario default blend exports to "skinned" (and the geo ends up being named skinned_geo and for some reason the collision exports to the actual armature name).
The upgrade code doesn´t and can´t account for defaults without being an annoyance, so maybe the mario blend needs a quick update for this pr too. (although the group not defaulting to group0 is a bit odd regardless)